phpunit/functional/Glpi/Inventory/InventoryTest.php

```diff
@@ -8454,4 +8454,285 @@ public function testPartialUser()
+        $count_vms = count($json->content->virtualmachines);
+        $nb_computers = countElementsInTable(\Computer::getTable());
+
+        $this->doInventory($json);
+
+        // Check created VMs
+        $this->assertSame($count_vms, countElementsInTable(\ItemVirtualMachine::getTable()));
+        $this->assertSame($nb_computers + 2, countElementsInTable(\Computer::getTable()));
+        $computer = new \Computer();
+        $computer->getFromDBByCrit(['name' => 'mailpit_update']);
+
+        // Create Docker VM type
+        $nb_vm = countElementsInTable(\VirtualMachineType::getTable());
+        $docker_vm = $this->createItems(
+            \VirtualMachineType::class,
+            [
+                ['name' => 'docker'],
+            ]
+        );
+        $this->assertGreaterThan($nb_vm, countElementsInTable(\VirtualMachineType::getTable()));
+
+        // Import rule to refuse "docker" virtual machine
+        $criteria = [
+            [
+                'condition' => 0,
+                'criteria'  => 'itemtype',
+                'pattern'   => 'Computer',
+            ], [
+                'condition' => \Rule::PATTERN_IS,
+                'criteria'  => 'virtualmachinetypes_id',
+                'pattern'   => $docker_vm[0]->fields['id']
+            ]
+        ];
+        $action = [
+            'action_type' => 'assign',
+            'field'       => '_ignore_import',
+            'value'       => \RuleImportAsset::RULE_ACTION_LINK_OR_NO_IMPORT
+        ];
+        $rule = new \RuleImportAsset();
+        $collection = new \RuleImportAssetCollection();
+        $rulecriteria = new \RuleCriteria();
+
+        $input = [
+            'is_active' => 1,
+            'name'      => 'Update virtualisation',
+            'match'     => 'AND',
+            'sub_type'  => 'RuleImportAsset',
+        ];
+
+        $rules_id = $rule->add($input);
+        $this->assertGreaterThan(0, $rules_id);
+        $this->assertTrue($collection->moveRule($rules_id, 0, $collection::MOVE_BEFORE));
+
+        // Add criteria
+        foreach ($criteria as $crit) {
+            $input = [
+                'rules_id'  => $rules_id,
+                'criteria'  => $crit['criteria'],
+                'pattern'   => $crit['pattern'],
+                'condition' => $crit['condition'],
+            ];
+            $this->assertGreaterThan(0, (int)$rulecriteria->add($input));
+        }
+
+        // Add action
+        $ruleaction = new \RuleAction();
+        $input = [
+            'rules_id'    => $rules_id,
+            'action_type' => $action['action_type'],
+            'field'       => $action['field'],
+            'value'       => $action['value'],
+        ];
+        $this->assertGreaterThan(0, (int)$ruleaction->add($input));
+
+        $nb_computers = countElementsInTable(\Computer::getTable());
+
+        // Second inventory
+        $json = json_decode(createJsonString("GLPI-Agent_v1.10-dev2"));
```

Comment:
It des not seems required to have a different agent; just using the same JSON should work the same.

Also, I do not really understand what the update really does from the VMs point of view. It checks nothing has changed; but I guess nothing would have changed without new rules added.